packages/plugin-ext/src/common/known-commands.ts

```diff
@@ -0,0 +1,329 @@
+/********************************************************************************
+ * Copyright (C) 2019 Red Hat, Inc. and others.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License v. 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0.
+ *
+ * This Source Code may also be made available under the following Secondary
+ * Licenses when the conditions for such availability set forth in the Eclipse
+ * Public License v. 2.0 are satisfied: GNU General Public License, version 2
+ * with the GNU Classpath Exception which is available at
+ * https://www.gnu.org/software/classpath/license.html.
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR GPL-2.0 WITH Classpath-exception-2.0
+ ********************************************************************************/
+
+import { Range as R, Position as P, Location as L } from 'vscode-languageserver-types';
+import URI from 'vscode-uri';
+import * as theia from '@theia/plugin';
+
+// Here is a mapping of VSCode commands to monaco commands with their conversions
+export namespace KnownCommands {
+
+    /**
+     * Commands that you want to apply custom conversions to rather than pass through the automatic args converter.
+     * Would be useful in the case where theia provides some command and you need to provide custom conversions
+     */
+    // tslint:disable-next-line:no-any
+    export const transformers: { [id: string]: [string, (args: any[] | undefined) => any[] | undefined] } = {};
```

Comment:
I think the internal storage structure should not be part of the public interface: what we want to here is 2 things:
1. Transform an invocation of a know VS Code command into an invocation of a known Theia command (including parameter conversions)
2. Map the id of a VS Code command to the id of a Theia command. 

So we just need map(...) and mappedId(...)


We should always strive to have minimal interfaces and to centralize behaviour as much as possible. Exposing the internal structure creates coupling of the users of this class to the _implementation_ of this class. This breaks easily.